src/Shared/ServerInfrastructure/Http2/ConnectionEndReason.cs

```diff
@@ -0,0 +1,58 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core;
+
+internal enum ConnectionEndReason
+{
+    Unset,
+    OtherError,
+    ConnectionReset,
+    FlowControlWindowExceeded,
+    KeepAliveTimeout,
+    InsufficientTlsVersion,
+    InvalidHandshake,
+    InvalidStreamId,
+    FrameAfterStreamClose,
+    UnknownStream,
+    UnsupportedFrame,
+    UnexpectedFrame,
+    InvalidFrameLength,
+    InvalidDataPadding,
+    InvalidRequestHeaders,
+    InvalidRequestLine,
```

Comment:
Seems like "Invalid" is usually a prefix.

Reply:
Did you put this comment on the wrong line?

I've changed the reasons to consistently start with "Invalid" where applicable.